logging/src/main/java/org/jboss/as/logging/LoggingModuleDependency.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2020 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.logging;
+
+import org.jboss.as.controller.registry.RuntimePackageDependency;
+
+/**
+ * Describes a module dependency.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public enum LoggingModuleDependency {
+    APACHE_COMMONS_LOGGING("org.apache.commons.logging"),
+    LOG4J("org.apache.log4j"),
+    LOG4J2("org.apache.logging.log4j.api"),
+    LOG4J2_IMPL("org.jboss.logmanager.log4j2", true),
```

Comment:
@jamezp I was wondering why you put in the LOG4J2_IMPL org.jboss.logmanager.lo4j2 as dependency. In [JCL](https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/common/src/main/resources/modules/system/layers/base/org/apache/commons/logging/main/module.xml) it is "This allows the org.jboss.logging.commons.logging module to be private while keeping this module public"
I think the Apache Commons Logging way, to provide the API but not the Impl is quite good. It prevents app developers to depend on that classes.

Reply:
If we don't add it here we have to add a circular dependency in the `org.apache.logging.log4j.api` module so it knows where to find the implementation. I suppose another option would be to create a `org.apache.logging.log4j` module and make the add that as a dependency to the deployment exporting only the API.

Reply:
Yes, you're right. I'll change my backports in the same way.